internal/http/services/owncloud/ocdav/get.go

```diff
@@ -124,7 +124,7 @@ func (s *svc) handleGet(ctx context.Context, w http.ResponseWriter, r *http.Requ
 
 	w.Header().Set(HeaderContentType, info.MimeType)
 	w.Header().Set(HeaderContentDisposistion, "attachment; filename*=UTF-8''"+
-		path.Base(info.Path)+"; filename=\""+path.Base(info.Path)+"\"")
+		path.Base(r.RequestURI)+"; filename=\""+path.Base(r.RequestURI)+"\"")
```

Comment:
@ishank011 the trick is to always take the filename from the request url.

Reply:
Yes we can keep the change but we shouldn't revert https://github.com/cs3org/reva/pull/2297, that'll break the case I described above

Reply:
we should probably cover this with a test @phil-davis

Reply:
@ishank011 could you check if https://github.com/owncloud/core/blob/762780a23c9eadda4fb5fa8db99eba66a5100b6e/tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature#L184-L221 covers the scenario you described? If not could you rephrase it in gherkin?

@phil-davis could you add a check for the content disposition header to the core testsuite?

Reply:
@butonic will take a look